src/components/themed/SwapInputCard.tsx

```diff
@@ -0,0 +1,295 @@
+import { div, log10, mul, round } from 'biggystring'
+import { EdgeCurrencyWallet, EdgeTokenId } from 'edge-core-js'
+import React, { useMemo } from 'react'
+import { ReturnKeyType, Text, TouchableOpacity, View } from 'react-native'
+
+import { useHandler } from '../../hooks/useHandler'
+import { useWatch } from '../../hooks/useWatch'
+import { lstrings } from '../../locales/strings'
+import { emptyEdgeDenomination, getExchangeDenom, selectDisplayDenom } from '../../selectors/DenominationSelectors'
+import { useSelector } from '../../types/reactRedux'
+import { getCurrencyCode } from '../../util/CurrencyInfoHelpers'
+import { DECIMAL_PRECISION, getDenomFromIsoCode, maxPrimaryCurrencyConversionDecimals, precisionAdjust } from '../../util/utils'
+import { styled } from '../hoc/styled'
+import { Space } from '../layout/Space'
+import { CryptoIconUi4 } from '../ui4/CryptoIconUi4'
+import { EdgeText } from './EdgeText'
+import { FieldNum, FlipInputFieldInfos, FlipInputNew, FlipInputRef } from './FlipInputNew'
+import { ButtonBox } from './ThemedButtons'
+
+export type ExchangeFlipInputFields = 'fiat' | 'crypto'
+
+export interface SwapInputCardInputRef {
+  setAmount: (field: ExchangeFlipInputFields, value: string) => void
+}
+
+export interface SwapInputCardAmounts {
+  exchangeAmount: string
+  nativeAmount: string
+  fiatAmount: string
+  fieldChanged: 'fiat' | 'crypto'
+}
+
+export interface Props {
+  heading?: string
+  wallet?: EdgeCurrencyWallet
```

Comment:
The `wallet` will always be passed, because you replace the whole card with a button when the wallet is `null`. This will have a whole bunch of clean-up effects, such as getting rid of the `useMaybeFiatCurrencyCode` hook, the `disabled` prop, and so forth.